### PR TITLE
Adding versioning to csm ldap schema

### PR DESCRIPTION
**Pull request by udayan-y21**
Signed-off-by: Udayan Yaragattikar <udayan.yaragattikar@seagate.com>

# Problem Statement
https://jts.seagate.com/browse/EOS-26141

# Design
- Add versioning in csm-ldap schema.
-  https://seagatetechnology.sharepoint.com/:p:/r/sites/gteamdrv1/tdrive1224/_layouts/15/doc2.aspx?sourcedoc=%7B337D2780-20C6-4863-8476-FD28DCDB5D1C%7D&file=OpenLDAP_schema_versioning.pptx&wdOrigin=OFFICECOM-WEB.START.SEARCH&ct=1637900172281&action=edit&mobileredirect=true&CID=8942f54e-bdc2-1cbf-d927-8299967390f6
- Finalized approch:
   2) dc=csm,dc=seagate,dc=com  ​
   dc=v1,dc=csm,dc=seagate,dc=com​
   dc=v2,dc=csm,dc=seagate,dc=com    ​
  ou=accounts,dc=v1,dc=csm,dc=seagate,dc=com
​
# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
Login to Cortxadmin user
image

Create another csm-user
image

Fetch csm users
image

**Ldap search command showing dc=v1 is added under dc=csm,dc=seagate,dc=com
Also Cortxadmin and csm_user01 user added under v1**
```
[root@cortx-control-headless-svc /]# ldapsearch -b "dc=csm,dc=seagate,dc=com" -x -w seagate2 -D "cn=authadmin,dc=seagate,dc=com" -H ldap://openldap-svc.default.svc.cluster.local:389
# extended LDIF
#
# LDAPv3
# base <dc=csm,dc=seagate,dc=com> with scope subtree
# filter: (objectclass=*)
# requesting: ALL
#

# csm.seagate.com
dn: dc=csm,dc=seagate,dc=com
dc: csm
o: csm
description: CORTX User directory.
objectClass: top
objectClass: dcObject
objectClass: organization

# v1.csm.seagate.com
dn: dc=v1,dc=csm,dc=seagate,dc=com
dc: v1
o: v1
description: CORTX User version v1 directory.
objectClass: top
objectClass: dcObject
objectClass: organization

# accounts, v1.csm.seagate.com
dn: ou=accounts,dc=v1,dc=csm,dc=seagate,dc=com
ou: accounts
objectClass: organizationalUnit

# userAccount, accounts, v1.csm.seagate.com
dn: o=userAccount,ou=accounts,dc=v1,dc=csm,dc=seagate,dc=com
o: userAccount
cortxAccountId: cortxUserAccount
objectClass: CortxAccount
mail: cortxUserAccount@seagate.com

# cortxadmin, userAccount, accounts, v1.csm.seagate.com
dn: user-id=cortxadmin,o=userAccount,ou=accounts,dc=v1,dc=csm,dc=seagate,dc=co
 m
objectClass: User
user-id: cortxadmin
user-type: csm
user-role: admin
user-password: $2b$12$KYm7XmPtViXCrN2G2URPr.G6zrVOV4X/tTLiphrk62QqaBLjMAB0.
email-address: cortx@seagate.com
reset-password: FALSE
alert-notification: TRUE
updated-time: 20211130154207Z
created-time: 20211130154207Z

# csm_user01, userAccount, accounts, v1.csm.seagate.com
dn: user-id=csm_user01,o=userAccount,ou=accounts,dc=v1,dc=csm,dc=seagate,dc=co
 m
objectClass: User
user-id: csm_user01
user-type: csm
user-role: monitor
user-password: $2b$12$7breQunjvxLjm9ZMWRrTcO4JayxXvRT8hHMXd/pXJEENSNtGHlflm
email-address: csm_user01@seagate.com
reset-password: FALSE
alert-notification: TRUE
updated-time: 20211201075523Z
created-time: 20211201075523Z

# search result
search: 2
result: 0 Success

# numResponses: 7
# numEntries: 6

```

S3 user-creation
image

S3-user Get
image

  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: csm/conf/configure.py [hunk 1]
```diff
@@ -372,7 +375,7 @@ def _configure_csm_ldap_schema(self):
                     self._create_csm_ldap_user(each_server_url)
                     # Setup necessary permissions
                     Log.info(f"Setup necessary permissions to {ldap_user} on server: {each_server_url}")
-                    self._setup_ldap_permissions(each_server_url, base_dn, ldap_user)
+                    self._setup_ldap_permissions(each_server_url, base_dn, ldap_user,csm_schema_version)
```
Comment (rohitkolapkar) [inline csm/conf/configure.py:378]: you may not need this param -> 'csm_schema_version' in  _setup_ldap_permissions(). please check.

**Review by rohitkolapkar — CHANGES_REQUESTED** [review]

**Comment by rohitkolapkar** [comment]
Other changes looks fine. Please test csm/S3 user management CRUD.

File: csm/conf/configure.py (hunk 1 above)
Comment (udayan-y21) [reply csm/conf/configure.py]: Done

**Review by rohitkolapkar — APPROVED** [review]

File: csm/conf/configure.py (hunk 1 above)
Comment (rohitkolapkar) [reply csm/conf/configure.py]: You can resolve this.

**Review by tb-s — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Pull request merged**